Monika After Story/game/0utils.rpy

```diff
@@ -1,3 +1,274 @@
+                kwargs["extra"] = new_extra
+            else:
+                kwargs["extra"] = dict(self.extra)
+
+            for prop_name in self.extra:
+                self._add_extra_prop(prop_name, kwargs)
+
+
+    class MASNewlineLogAdapter(MASExtraPropLogAdapter):
+        """
+        Log adapter with an option for newline kwargs.
+        The newline kwarg is pfx_newline.
+        """
+
+        def __init__(self, logger, extra_props=None, newline_def=False):
+            """
+            IN:
+                logger - the logger to adapt
+                extra_props - additional props, other than the newline one.
+                    Optional.
+                    (Default: None)
+                newline_def - the default value for the newline prop
+                    (Default: False)
+            """
+            if extra_props is None:
+                extra_props = {}
+
+            extra_props["pfx_newline"] = newline_def
+            super(MASNewlineLogAdapter, self).__init__(logger, extra_props)
+
+
+    #We always log to renpy.config.basedir/log
+    LOG_PATH = os.path.join(renpy.config.basedir, "log")
+
+    LOG_MAXSIZE_B = 5242880 #5 mb
+
+    #Add the header to each log, including OS info + MAS version number
+    LOG_HEADER = "\n\n{0}\n{1}\n{2}\n\nVERSION: {3}\n{4}"
+
+    #Unformatted logs use these consts (spj/pnm)
+    MSG_INFO = "[" + LT_INFO + "]: {0}"
+    MSG_WARN = "[" + LT_WARN + "]: {0}"
+    MSG_ERR = "[" + LT_ERROR + "]: {0}"
+
+    MSG_INFO_ID = "    " + MSG_INFO
+    MSG_WARN_ID = "    " + MSG_WARN
+    MSG_ERR_ID = "    " + MSG_ERR
+
+    #Load strs for files
+    LOAD_TRY = "Attempting to load '{0}'..."
+    LOAD_SUCC = "'{0}' loaded successfully."
+    LOAD_FAILED = "Load failed."
+
+    JSON_LOAD_FAILED = "Failed to load json at '{0}'."
+    FILE_LOAD_FAILED = "Failed to load file at '{0}'. | {1}"
+    NAME_BAD = "name must be unique."
+
+    #Ensure log path exists
+    if not os.path.exists(LOG_PATH):
+        os.makedirs(LOG_PATH)
+
+    #Full logging info
+    def init_log(name, append=True, formatter=None, adapter=None):
+        """
+        Initializes a logger with a handler with the name and files given.
+
+        IN:
+            name - name of the logger, this will be the same as the file, with the file appending '.txt'
+            append - Whether or not we're appending this log or clearing it on load
+                (Default: True)
+            formatter - custom logging.Formatter to be used.
+                If None is provided, the default MASLogFormatter is used.
+                (Default: None)
+            adapter - Constructor reference to the adapter we want to use. If None, no adapter is used
+                (Default: None)
+
+        NOTE: ALL LOGS ARE IN renpy.config.basedir/log/
+        All logs flush and rotate once they're 5 mb in size.
+        """
+        #NOTE: using `delay` causes weird issues where the filestream is nonexistent in renpy. Do not use it
```

Comment:
I tested it and `delay` works. The error was somewhere else and it's fixed, apparently. `delay` can be good for optional logs.
```suggestion
```

Reply:
Works with 7.1, can't test 7.4